bld/CLMBuildNamelist.pm

```diff
@@ -3593,68 +3692,49 @@ sub setup_logic_popd_streams {
-                  'sim_year'=>$nl_flags->{'sim_year'},
-                  'sim_year_range'=>$nl_flags->{'sim_year_range'});
-      add_default($opts, $nl_flags->{'inputdata_rootdir'}, $definition, $defaults, $nl, 'stream_year_last_lightng',
-                  'sim_year'=>$nl_flags->{'sim_year'},
-                  'sim_year_range'=>$nl_flags->{'sim_year_range'});
-      # Set align year, if first and last years are different
-      if ( $nl->get_value('stream_year_first_lightng') !=
-           $nl->get_value('stream_year_last_lightng') ) {
-        add_default($opts, $nl_flags->{'inputdata_rootdir'}, $definition, $defaults, $nl, 'model_year_align_lightng', 'sim_year'=>$nl_flags->{'sim_year'},
-                    'sim_year_range'=>$nl_flags->{'sim_year_range'});
-     }
-     add_default($opts, $nl_flags->{'inputdata_rootdir'}, $definition, $defaults, $nl, 'stream_fldfilename_lightng',
-                 'hgrid'=>$nl_flags->{'light_res'} );
+  if ( $nl_flags->{'light_res'} ne "none" ) {
+      my $inputdata_rootdir = $nl_flags->{'inputdata_rootdir'};
+      my ($var_nml, $var_opts)  = ('stream_year_first_lightng', 'stream_lightng_year_first');
+      add_stream_default($opts, $inputdata_rootdir, $definition, $defaults, $nl, $var_nml, $var_opts, 0);
+      my ($var_nml, $var_opts)  = ('stream_year_last_lightng', 'stream_lightng_year_last');
+      add_stream_default($opts, $inputdata_rootdir, $definition, $defaults, $nl, $var_nml, $var_opts, 0);
+      my ($var_nml, $var_opts)  = ('model_year_align_lightng', 'stream_lightng_year_align');
+      add_stream_default($opts, $inputdata_rootdir, $definition, $defaults, $nl, $var_nml, $var_opts, 0);
+      my ($var_nml, $var_opts)  = ('stream_fldfilename_lightng', 'stream_lightng_data_filename');
+      add_stream_default($opts, $inputdata_rootdir, $definition, $defaults, $nl, $var_nml, $var_opts, 1);
       if ($opts->{'driver'} eq "nuopc" ) {
-          add_default($opts, $nl_flags->{'inputdata_rootdir'}, $definition, $defaults, $nl, 'stream_meshfile_lightng',
-                      'hgrid'=>$nl_flags->{'light_res'} );
+          my ($var_nml, $var_opts)  = ('stream_meshfile_lightng', 'stream_lightng_mesh_filename');
+          add_stream_default($opts, $inputdata_rootdir, $definition, $defaults, $nl, $var_nml, $var_opts, 1);
       }
-  } else {
-     # If bgc is NOT CN/CNDV then make sure none of the Lightng settings are set
-     if ( defined($nl->get_value('stream_year_first_lightng')) ||
-          defined($nl->get_value('stream_year_last_lightng'))  ||
-          defined($nl->get_value('model_year_align_lightng'))  ||
-          defined($nl->get_value('lightng_tintalgo'        ))  ||
-          defined($nl->get_value('stream_fldfilename_lightng'))   ) {
-        $log->fatal_error("When bgc is SP (NOT CN or BGC or FATES) or fire is turned off none of: stream_year_first_lightng,\n" .
```

Comment:
This removes a lot of the error checking which I really don't think should be done here. That ensures that the user isn't changing things that won't have any affect on their specific case. A real downside of this is that you can freely change the CLM_STREAMS_* settings for variables that would have no affect on your case and not know about it.